app/controllers/UploadFormController.scala

```diff
@@ -90,21 +94,26 @@ class UploadFormController @Inject()(
       }
     }
 
-  def showError(errorCode: String, errorMessage: String, errorRequestId: String): Action[AnyContent] = Action.async {
-    implicit request => errorCode match {
-      case "EntityTooLarge" =>
-        renderer.render(
-          "fileTooLargeError.njk",
-          Json.obj("xmlTechnicalGuidanceUrl" -> Json.toJson(appConfig.xmlTechnicalGuidanceUrl))
-        ).map (Ok (_))
-      case _ =>
+  def showError(errorCode: String, errorMessage: String, errorRequestId: String): Action[AnyContent] = (identify andThen getData).async {
+    implicit request =>
+
+      errorCode match {
+        case "EntityTooLarge" =>
+          renderer.render(
+            "fileTooLargeError.njk",
+            Json.obj("xmlTechnicalGuidanceUrl" -> Json.toJson(appConfig.xmlTechnicalGuidanceUrl))
+          ).map (Ok (_))
+        case "InvalidArgument" =>
+          val formWithErrors: Form[String] = form.withError("file", "upload_form.error.file.empty")
+          toResponse(formWithErrors)
```

Comment:
Please check after implementing suggested changes do we still need this block of code